app/routes/events/view/edit.js

```diff
@@ -12,6 +12,14 @@ export default Route.extend(AuthenticatedRouteMixin, EventWizardMixin, {
     if (transition.targetName === 'events.view.edit.index') {
       this.transitionTo('events.view.edit.basic-details');
     }
+
+    let event = this.modelFor('events.view');
+    let { currentUser } = this.authManager;
+    if (!(currentUser.isAnAdmin || currentUser.email === event.owner.get('email') || event.organizers.includes(currentUser)
+        || event.coorganizers.includes(currentUser) || event.trackOrganizers.includes(currentUser)
```

Comment:
@shreyanshdwivedi Just make sure if this hierarchy correct, I mean not every `event role` should have access to `Edit Page`. Confirm with @CosmicCoder96 or @mariobehling once, Else it is gtg.

Reply:
I think currently we are giving full access to all the event roles. @CosmicCoder96 is there any role which is needed to be removed?